src/MergeContext.js

```diff
@@ -906,6 +955,38 @@ class PullRequest {
         return this._rawPr.body.replace(/\r+\n/g, '\n');
     }
 
+    _preprocessedPrBody() {
+        assert(this._preprocessedPrBodyCache !== undefined);
+        return this._preprocessedPrBodyCache;
+    }
+
+    _parseAuthor(attr, str) {
+        const cred = str.match(/^([\w][^@<>,]*) <(\S+@\S+\.\S+)>$/);
+        if (!cred) {
+            this._warn(`Invalid '${attr}' line format`);
+            return null;
+        }
+        if (cred[0].includes(',')) {
+            this._warn(`Invalid '${attr}' line: commas are not allowed`);
+            return null;
+        }
+        const now = new Date();
+        return {name: cred[1].trim(), email: cred[2].trim(), date: now.toISOString()};
+    }
```

Comment:
I do not think we should risk this ephemeral `date` field getting into the repository. I am guessting that you wanted the structure of the returned author object to match the structure of the _stagedCommit.author, but since only one of them can contain valid dates, I think it is too dangerous to do that. The two structures should not be viewed as compatible. While _stagedCommit.author comes from GitHub, we can create a basic NameMail class to represent the Authored-by and Co-authored-by information (just name and email, no date).

This leaves the question of setting the right AuthorDate for staged commits of Authored-by PRs. I think that date should be the same as it would have been if Authored-by did not exist. In other words, adding or changing an Authored-by header should not change the AuthorDate field of the merged commit.

Reply:
I removed the date field - the date value is now provided from the merge commit (i.e., the date used with the default author). I do not think we should overcomplicate with an additional NameMail (at least for now) - just returning the unnamed hash (with two fields) should be enough.